api/client.ts

```diff
@@ -269,7 +281,7 @@ class Client {
       promise will not be resolved or rejected if a callback is provided.
   */
   async login(
-    credentials: Credentials,
+    credentials?: Credentials,
```

Comment:
Was wondering if putting the optional param in `login` and `connectAndLogin` at the end of the params list might be a better practice? This way, when not available, we would just not pass the param at the end instead of passing it as `undefined` in the middle.

Reply:
In both cases `clientVersion` is also optional, but with a default value (in TS you can't mark a param as optional and provide a default value as the variable will always be initialised). Incidentally, having an optional param before a non-optional gives a TS error.

Also, if we were to swap the order it would mean we would always have to provide the client version when we wanted to pass credentials, which would mean we wouldn't get the benefit of having a default value.

Reply:
Makes sense. I didn't spot that `clientVersion` is provided a default value.